iothub/device/src/IotHubModuleClient.cs

```diff
@@ -198,7 +198,7 @@ public async Task SendTelemetryBatchAsync(string outputName, IEnumerable<Telemet
         /// <returns>The result of the method invocation.</returns>
         /// <exception cref="InvalidOperationException">Thrown if IotHubModuleClient instance is not opened already.</exception>
         /// <exception cref="OperationCanceledException">Thrown when the operation has been canceled.</exception>
-        public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, DirectMethodRequest methodRequest, CancellationToken cancellationToken = default)
+        public Task<DirectMethodResponse> InvokeMethodAsync(string deviceId, DirectMethodServiceRequest methodRequest, CancellationToken cancellationToken = default)
```

Comment:
Hmm, true. Ideas?

Reply:
Maybe leave this name alone and change it on the service client side instead?

Reply:
```csharp
// device/module client side
public class DirectMethodRequest; // can be sent from a module or from the service
public class DirectMethodResponse; // can be sent back to either a module or the service

//service client side
public class DirectMethodServiceRequest; // can only be sent by the service, but is functionally the same as DirecMethodRequest otherwise
public class DirectMethodClientResponse; // can only be returned by the device/module (via the service)
```

Reply:
`DirectMethodExternalRequest`?

Reply:
FWIW, I don't necessarily mind the "service" name here. I do think of Edge Hub as part of the service.